src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java

```diff
@@ -618,14 +618,69 @@ Field buildField(ColumnMetadata column) {
 
     // Create the corresponding column field base on the column data type
     fieldType = new FieldType(column.getNullable(), arrowType, null, metadata);
-    return new Field(column.getName(), fieldType, children);
+    String columnName = unquote(column.getName());
+    return new Field(columnName, fieldType, children);
   }
 
   private String formatColumnName(String columnName) {
     Utils.assertStringNotNullOrEmpty("invalid column name", columnName);
-    return (columnName.charAt(0) == '"' && columnName.charAt(columnName.length() - 1) == '"')
-        ? columnName
-        : columnName.toUpperCase();
+    if (columnName.charAt(0) == '"' && columnName.charAt(columnName.length() - 1) == '"') {
```

Comment:
What about `"""`?

Reply:
hmm I cannot find any special case for `"""` in server side code and [docs](https://docs.snowflake.com/en/sql-reference/identifiers-syntax.html). This will handled in `unquote()` depending on where it is as quoted or double-quote escape.
could it be that you are talking about const string expressions?